### PR TITLE
Railsconf 2023 videos

### PR DESCRIPTION
**Pull request by PedroAugustoRamalhoDuarte**
First of all thanks for the project, it was so easy to setup.

This PRs adds rails conf 2023 videos, it still work on progress, and i am developing a better parser. 

```ruby
require "json"
require "yaml"

# This scripts get the json file from youtube api and parser to yaml file in rubyvideo format
# This scripts does not handle well:
# - Keynotes
# - Multiples speakers that are not separated by ","


# Get JSON File from: https://developers.google.com/youtube/v3/docs/playlistItems/list
json = File.read("./test.json")
playlist = JSON.parse(json)

videos = playlist["items"].map do |video|

  # Handle speakers
  speakers = video["snippet"]["title"].split("by")[-1]
  speakers = speakers.split(",").map(&:strip).map do |speaker|
    speaker
  end

  raw_title = video["snippet"]["title"]

  {
    "title" => raw_title.split(" by ")[0].gsub("RailsConf 2023 -", "").strip,
    "raw_title" => raw_title,
    "speakers" => speakers,
    "description" => video["snippet"]["description"],
    "thumbnail_xs" => video["snippet"]["thumbnails"]["default"]["url"],
    "thumbnail_sm" => video["snippet"]["thumbnails"]["medium"]["url"],
    "thumbnail_md" => video["snippet"]["thumbnails"]["high"]["url"],
    "thumbnail_lg" => video["snippet"]["thumbnails"]["standard"]["url"],
    "thumbnail_xl" => video["snippet"]["thumbnails"]["maxres"]["url"],
    "published_at" => video["snippet"]["publishedAt"].split("T")[0],
    "video_id" => video["snippet"]["resourceId"]["videoId"]
  }
end

pp videos.length

yaml_output = videos.to_yaml

# Write YAML to a file
File.open('output.yaml', 'w') { |file| file.write(yaml_output) }
```

### CODE REVIEW

**Comment by adrienpoly** [comment]
Thanks for your work I am late cleaning up and publishing my scripts I used for the initial batch. Hope to get this done by the end of the week. Anyway that won't prevent us to move forward with RailsConf23 vids

If you want to bring it to the finish line before that here are a few things to correct:

- remove "RailsConf 2023 -" from the title. This is very repetitive and information we already have displayed on the card (event name). Except when it is a keynote as the keynote usually doesn't have a full title so here I feel it makes sense to key the RailsConf23 in the title
- remove the speakers from the title. Given this raw title `RailsConf 2023 - 10x your teamwork through pair programming by Michael Milewski, Selena Small` expected title is `10x your teamwork through pair programming`
- manually check the speaker for the keynotes. usually, the pattern for keynotes is different from other titles and the speakers are not always well extracted. Here Aaron is not well extracted as an example

thanks again for taking this task!

**Comment by PedroAugustoRamalhoDuarte** [comment]
@adrienpoly i have fixed the title,  manually fixed some speakears and add the last videos.

File: data/railsconf/railsconf-2023/videos.yml [hunk 1]
```diff
@@ -0,0 +1,1044 @@
...
+    PWA, taking advantage of the new front-end tools that come with Rails by default
+    since version 7. Entry-level developers are welcome!
+  thumbnail_xs: https://i.ytimg.com/vi/4cQuxQdWkPE/default.jpg
+  thumbnail_sm: https://i.ytimg.com/vi/4cQuxQdWkPE/mqdefault.jpg
+  thumbnail_md: https://i.ytimg.com/vi/4cQuxQdWkPE/hqdefault.jpg
+  thumbnail_lg: https://i.ytimg.com/vi/4cQuxQdWkPE/sddefault.jpg
+  thumbnail_xl: https://i.ytimg.com/vi/4cQuxQdWkPE/maxresdefault.jpg
+  published_at: '2023-07-10'
+  video_id: 4cQuxQdWkPE
+- title: Accessible
```
Comment (marcoroth) [inline data/railsconf/railsconf-2023/videos.yml:53]: Looks like we are missing half of the title here
```suggestion
- title: Accessible by default
```

File: data/railsconf/railsconf-2023/videos.yml [hunk 2]
```diff
@@ -0,0 +1,1044 @@
...
+
+    When choosing to follow a modular approach, I explored applying microservices patterns to our modular monolith. In this talk, I’ll share my experience in adopting this approach, focusing on ensuring data consistency and distributed transactions, as well as the pitfalls and insights I discovered along the way.
+  thumbnail_xs: https://i.ytimg.com/vi/4zrQAJ0RlI4/default.jpg
+  thumbnail_sm: https://i.ytimg.com/vi/4zrQAJ0RlI4/mqdefault.jpg
+  thumbnail_md: https://i.ytimg.com/vi/4zrQAJ0RlI4/hqdefault.jpg
+  thumbnail_lg: https://i.ytimg.com/vi/4zrQAJ0RlI4/sddefault.jpg
+  thumbnail_xl: https://i.ytimg.com/vi/4zrQAJ0RlI4/maxresdefault.jpg
+  published_at: '2023-07-10'
+  video_id: 4zrQAJ0RlI4
+- title: 'Bridging the Gap: Creating Trust Between Non-Technical...'
```
Comment (marcoroth) [inline data/railsconf/railsconf-2023/videos.yml:102]: It also looks like all titles ending in `...` are cut off in the YouTube title itself
```suggestion
- title: Bridging the Gap: Creating Trust Between Non-Technical Stakeholders and Engineering Teams
```

File: data/railsconf/railsconf-2023/videos.yml [hunk 3]
```diff
@@ -0,0 +1,1044 @@
...
+    development, and an accessibility-first form builder to make GitHub's 15-year-old,
+    1,400-controller Ruby on Rails monolith accessible.
+  thumbnail_xs: https://i.ytimg.com/vi/4j2zlvE_Yj8/default.jpg
+  thumbnail_sm: https://i.ytimg.com/vi/4j2zlvE_Yj8/mqdefault.jpg
+  thumbnail_md: https://i.ytimg.com/vi/4j2zlvE_Yj8/hqdefault.jpg
+  thumbnail_lg: https://i.ytimg.com/vi/4j2zlvE_Yj8/sddefault.jpg
+  thumbnail_xl: https://i.ytimg.com/vi/4j2zlvE_Yj8/maxresdefault.jpg
+  published_at: '2023-07-10'
+  video_id: 4j2zlvE_Yj8
+- title: ActiveRecord::Encryption; Stop Hackers from...
```
Comment (marcoroth) [inline data/railsconf/railsconf-2023/videos.yml:69]:
```suggestion
- title: ActiveRecord::Encryption; Stop Hackers from Reading your Data
```

File: data/railsconf/railsconf-2023/videos.yml [hunk 4]
```diff
@@ -0,0 +1,1044 @@
...
+
+    In this talk, we'll explore how to build trust between non-technical stakeholders and engineering teams, using data and analysis to effectively communicate progress. We'll discuss practical tactics for leveraging the Pivotal Tracker API and Google Apps Scripts to extract relevant metrics, analyze data, and provide insightful analysis, as well as how to communicate that information. Join us to learn how to build trust, and how it can improve project success and create better outcomes for all parties involved.
+  thumbnail_xs: https://i.ytimg.com/vi/5gSRKyuO7qY/default.jpg
+  thumbnail_sm: https://i.ytimg.com/vi/5gSRKyuO7qY/mqdefault.jpg
+  thumbnail_md: https://i.ytimg.com/vi/5gSRKyuO7qY/hqdefault.jpg
+  thumbnail_lg: https://i.ytimg.com/vi/5gSRKyuO7qY/sddefault.jpg
+  thumbnail_xl: https://i.ytimg.com/vi/5gSRKyuO7qY/maxresdefault.jpg
+  published_at: '2023-07-10'
+  video_id: 5gSRKyuO7qY
+- title: 'Forecasting the Future: An Introduction to Machine Learning for...'
```
Comment (marcoroth) [inline data/railsconf/railsconf-2023/videos.yml:118]:
```suggestion
- title: Forecasting the Future: An Introduction to Machine Learning for Weather Prediction
```

File: data/railsconf/railsconf-2023/videos.yml [hunk 5]
```diff
@@ -0,0 +1,1044 @@
...
+
+    Join us for a town hall-style event where we'll review what each rubocop-rails rule does, hear its pros and cons, and put it to a vote. We'll enshrine the consensus picks in the first public release of standard-rails. By showing up, you'll not only be establishing conventions to promote safe and consistent code, you'll be sharpening the focus of Rails developers around the world!
+  thumbnail_xs: https://i.ytimg.com/vi/QVilOzkLdlI/default.jpg
+  thumbnail_sm: https://i.ytimg.com/vi/QVilOzkLdlI/mqdefault.jpg
+  thumbnail_md: https://i.ytimg.com/vi/QVilOzkLdlI/hqdefault.jpg
+  thumbnail_lg: https://i.ytimg.com/vi/QVilOzkLdlI/sddefault.jpg
+  thumbnail_xl: https://i.ytimg.com/vi/QVilOzkLdlI/maxresdefault.jpg
+  published_at: '2023-07-10'
+  video_id: QVilOzkLdlI
+- title: Building a world class engineering organization — learning from...
```
Comment (marcoroth) [inline data/railsconf/railsconf-2023/videos.yml:391]:
```suggestion
- title: Building a world class engineering organization — learning from cave
     paintings and horsey land art
```

File: data/railsconf/railsconf-2023/videos.yml [hunk 6]
```diff
@@ -0,0 +1,1044 @@
...
+    we knew for certain that our Rails apps would be in use in ten years (or a hundred!),
+    what would future us wish that we would've done now to build a better future?"
+  thumbnail_xs: https://i.ytimg.com/vi/R7JqiVPuKD8/default.jpg
+  thumbnail_sm: https://i.ytimg.com/vi/R7JqiVPuKD8/mqdefault.jpg
+  thumbnail_md: https://i.ytimg.com/vi/R7JqiVPuKD8/hqdefault.jpg
+  thumbnail_lg: https://i.ytimg.com/vi/R7JqiVPuKD8/sddefault.jpg
+  thumbnail_xl: https://i.ytimg.com/vi/R7JqiVPuKD8/maxresdefault.jpg
+  published_at: '2023-07-10'
+  video_id: R7JqiVPuKD8
+- title: 'Terms of Deployment: The Process of Evaluating Hatchbox, Fly and...'
```
Comment (marcoroth) [inline data/railsconf/railsconf-2023/videos.yml:410]:
```suggestion
- title: Terms of Deployment: The Process of Evaluating Hatchbox, Fly and Render for Developers
```

**Comment by adrienpoly** [comment]
Thanks @PedroAugustoRamalhoDuarte and @marcoroth I merged it and discovered a few things to fix that I pushed to main. Will try to deploy it this afternoon.

**Pull request merged**